### DISCUSSION TITLE
NPC Enhancements

### DISCUSSION THREAD

**Discussion by froggytrav** [AscensionGameDev/Intersect-Engine#1523] [Ideas]
I would like to make a few recommendations to NPCs.

1. NPC tab in Editor, select Move Route (same as event move route) for selected NPC.
2. in NPC Editor, 'Swarm with' box, and a set range. For example:
NPC A swarms with NPC B, NPC C, etc, with a detection range of 12, so if you aggro NPC A while it is within 12 range of selected NPCs, they will aggro along with A. Or in general just some way to link NPCs together.
3. Friendly casts, e.g NPCs to cast spells (healing, buffs, cleanse etc) on other NPCs, probably through a 'Friendly Target' list in the NPC Editor?
4. Trigger common event on NPC %hp
5. Set Projectile attack for NPCs (like Weapons in the Item Editor)
6. Connect Paperdoll to NPC
7. NPCs to be able to cast extra effects on themselves (Barrier, Invulnerable etc)

**Additional context**
Overall NPCs have received a few great additions, but there is some lacking aspects to NPC combat that would greatly contribute to a better gameplay experience.

**Comment by WeylonSantana** [comment]
1 - Asking the npc to follow the route is the same as trying to ask for the pet system, which could be done, in a very uncool way, but it would be nice for other purposes
2 - already exists? The only thing that doesn't exist is configurable distance
![image](https://user-images.githubusercontent.com/63019821/192142670-9ff3f996-1eea-4d70-b19b-607fc912d427.png)
3 - is nice
4 - also nice
5 - wonderful
6 - about this I dont know, maybe could be cool
7 - It's already possible

3, 4, 5 - okay
2 - parcially exist
7 - exist
1, 6 - idk how good it is
